pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,276 @@ type DiscoveryReconciler struct {
+		klog.Infof("Flavour found: %s", flavour.FlavourID)
+		cr := createFlavourCustomResource(flavour)
+		flavoursCR = append(flavoursCR, cr)
+	}
+
+	klog.Info("Flavours created", flavoursCR)
+	return flavoursCR, nil
+}
+
+// createFlavourCustomResource creates a Flavour CR from a Flavour
+func createFlavourCustomResource(flavour *nodecorev1alpha1.FlavourSpec) *nodecorev1alpha1.Flavour {
+	flavourCR := (&nodecorev1alpha1.Flavour{
+		// There is a warning: unknown field "spec.flavour.metadata.creationTimestamp"
+		// No issues, it is caused by kubebuilder for the creationTimestamp field, common bug for this version of k8s
+		Spec: nodecorev1alpha1.FlavourSpec{
+			FlavourID:  flavour.FlavourID,
+			ProviderID: flavour.ProviderID,
+			Type:       nodecorev1alpha1.K8S,
+			Characteristics: nodecorev1alpha1.Characteristics{
+				Cpu:    flavour.Characteristics.Cpu,
+				Memory: flavour.Characteristics.Memory,
+			},
+			Policy: nodecorev1alpha1.Policy{
+				// Check if flavour.Partitionable is not nil before setting Partitionable
+				Partitionable: func() *nodecorev1alpha1.Partitionable {
+					if flavour.Policy.Partitionable != nil {
+						return &nodecorev1alpha1.Partitionable{
+							CpuMin:     flavour.Policy.Partitionable.CpuMin,
+							MemoryMin:  flavour.Policy.Partitionable.MemoryMin,
+							CpuStep:    flavour.Policy.Partitionable.CpuStep,
+							MemoryStep: flavour.Policy.Partitionable.MemoryStep,
+						}
+					}
+					return nil
+				}(),
+				Aggregatable: func() *nodecorev1alpha1.Aggregatable {
+					if flavour.Policy.Aggregatable != nil {
+						return &nodecorev1alpha1.Aggregatable{
+							MinCount: flavour.Policy.Aggregatable.MinCount,
+							MaxCount: flavour.Policy.Aggregatable.MaxCount,
+						}
+					}
+					return nil
+				}(),
+			},
+			Owner: nodecorev1alpha1.NodeIdentity{
+				Domain: flavour.Owner.Domain,
+				IP:     flavour.Owner.IP,
+				NodeID: flavour.Owner.NodeID,
+			},
+			Price: nodecorev1alpha1.Price{
+				Amount:   flavour.Price.Amount,
+				Currency: flavour.Price.Currency,
+				Period:   flavour.Price.Period,
+			},
+			OptionalFields: nodecorev1alpha1.OptionalFields{
+				Availability: flavour.OptionalFields.Availability || false,
+			},
+		},
+		Status: nodecorev1alpha1.FlavourStatus{
+			LastUpdateTime: time.Now().String(),
+		},
+	})
+
+	return flavourCR
+}
+
+// createPeeringCandidateCustomResources creates a PeeringCandidate CR from a Flavour and a Discovery
+func createPeeringCandidateCustomResources(flavourPeeringCandidate *nodecorev1alpha1.Flavour, discovery *advertisementv1alpha1.Discovery, reserved bool, seq int) *advertisementv1alpha1.PeeringCandidate {
+	peeringCandidate := (&advertisementv1alpha1.PeeringCandidate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "peeringcandidate" + fmt.Sprintf("%d", seq),
+			Namespace: "default",
+		},
+		Spec: advertisementv1alpha1.PeeringCandidateSpec{
+			Reserved: reserved,
+			SolverID: discovery.Spec.SolverID,
+			Flavour:  *flavourPeeringCandidate,
```

Comment:
Let's discuss about it, if i change here the logic I have to change the logic everywhere and it would be a HUGE mess.

Reply:
not sure it will be so breaker

Reply:
AFTER CALL: as we discuss, for now we do not use a GenericRef in a PeeringCandidate (because the namespace in the PC could not be the same on the other sides).